src/Servers/Kestrel/Core/src/Adapter/Internal/AdaptedPipeline.cs

```diff
@@ -134,6 +141,11 @@ private async Task ReadInputAsync(Stream stream)
                     }
                 }
             }
+            catch (OperationCanceledException ex)
+            {
+                // Propagate the exception if it's ConnectionAbortedException
+                error = ex as ConnectionAbortedException;
```

Comment:
Why not make this an empty `catch (ConnectionAbortedException) { }` block?

Reply:
I misread. You are only storing this if it *is* a ConnectionAbortedException. In that case having a non-empty ConnectionAbortedException catch then an empty OperationCanceledException catch after is still probably better.

Reply:
This is less code.